src/main/kotlin/de/bigboot/ggtools/fang/service/MistforgeService.kt

```diff
@@ -90,21 +112,37 @@ class MistforgeService: AutostartService, KoinComponent {
                 4 -> 4
                 else -> 5
             }
-
+            
             val icon = ImmutableImage.loader()
-                .fromBytes(mistforge.getHeroImage(guide.heroId, "${name}.png").bytes())
+                .fromBytes(mistforge.getHeroImage(heroId, "${name}.png").bytes())
                 .scaleTo(96, 96, ScaleMethod.Bicubic)
 
             img = img.overlay(icon, x, y)
         }
-
+        
+        var talentId = 0
+        if (guide != null) {
+            talentId = guide.talent
+        }
+        else if (query_pairs.containsKey("talent")) {
+            talentId = query_pairs.get("talent")!!.toInt()
+        }
+        
         val talent = ImmutableImage.loader()
-            .fromBytes(mistforge.getHeroImage(guide.heroId, "${guide.talent+5}.png").bytes())
+            .fromBytes(mistforge.getHeroImage(heroId, "${talentId+5}.png").bytes())
             .scaleTo(96, 96, ScaleMethod.Bicubic)
 
         img = img.overlay(talent, 1117,262)
 
-        for ((i, c) in guide.skills.withIndex())
+        var skills = "eeeeeeeeee"
```

Comment:
use `when` instead of this if/else thing, also rather return without parsing the url than setting some nonsense default